packages/office-ui-fabric-react/src/components/Callout/Callout.test.tsx

```diff
@@ -141,4 +141,72 @@ describe('Callout', () => {
     focusTarget.focus();
     expect(gotEvent).toEqual(true);
   });
+
+  it('It will correctly return focus to element that spawned it', () => {
+    jest.useFakeTimers();
+    let visible = true;
+    let threwException = false;
+    let previousFocusElement;
+    let isFocused;
+    let restoreCalled = false;
+    const onRestoreFocus = (options: { originalElement: any; containsFocus: any }) => {
+      previousFocusElement = options.originalElement;
+      isFocused = options.containsFocus;
+      restoreCalled = true;
+    };
+
+    const onDismiss = () => {
+      visible = false;
+    };
+
+    // In order to have eventlisteners that have been added to the window to be called the JSX needs
+    // to be rendered into the real dom rather than the testutil simulated dom.
+    const root = document.createElement('div');
+    document.body.appendChild(root);
+    try {
+      ReactDOM.render<HTMLDivElement>(
+        <div>
```

Comment:
why not use enzyme `mount`? Does it create a problem?

Reply:
The problem here is that if this test were to fail, say on an assert, it leaves stuff in the dom. That means if another test ran looking for `#target` or `#focustarget` selectors, they would get the wrong element. It's important to try and leave the test environment clean when you're done and not leaking stuff to avoid side effects. You could just for example have a try/finally, or an `afterTest` callback, which unmounts and removes the root div if it exists. But I think enzyme mount tries to scope the render in its own little area. I think... i might be wrong :)

Reply:
I'll add an after that cleans this.